Lib/test/test_type_params.py

```diff
@@ -1148,3 +1160,103 @@ class Inner[U](T):
         """
         with self.assertRaises(RuntimeError):
             run_code(code)
+
+
+class DefaultsTest(unittest.TestCase):
+    def test_defaults_on_func(self):
+        ns = run_code("""
+            def func[T=int, **U=float, *V=None]():
+                pass
+        """)
```

Comment:
An alternative solution is just to add this file to this list here of files excluded from Ruff for now:

https://github.com/python/cpython/blob/eb927e9fc823de9539fcb82c9ea9d055462eb04a/Lib/test/.ruff.toml#L2-L20

We could probably remove it again fairly soon; the necessary updates to Ruff's parser probably won't take _too_ long? But I think this is also fine, and it's nice to keep the list of files excluded from Ruff as small as possible.

Reply:
I considered that but it's actually sort of good to put all the new syntax in strings, so if I mess up the compiler, it's easier to isolate the failure.